core/src/main/scala/com/salesforce/op/stages/impl/feature/MathTransformers.scala

```diff
@@ -221,3 +221,174 @@ class ScalarDivideTransformer[I <: OPNumeric[_], N]
+}
+
+
+/**
+ * Floor transformer
+ *
+ * @param uid      uid for instance
+ * @param tti      type tag for input
+ * @tparam I       input feature type
+ */
+class FloorTransformer[I <: OPNumeric[_]]
+(
+  uid: String = UID[FloorTransformer[_]]
+)(
+  implicit override val tti: TypeTag[I]
+) extends UnaryTransformer[I, Integral](operationName = "floor", uid = uid){
+  override def transformFn: I => Integral = (i: I) => i.toDouble.map(v => math.round(math.floor(v))).toIntegral
+}
+
+
+/**
+ * Round transformer
+ *
+ * @param uid      uid for instance
+ * @param tti      type tag for input
+ * @tparam I       input feature type
+ */
+class RoundTransformer[I <: OPNumeric[_]]
+(
+  uid: String = UID[FloorTransformer[_]]
+)(
+  implicit override val tti: TypeTag[I]
+) extends UnaryTransformer[I, Integral](operationName = "floor", uid = uid){
+  override def transformFn: I => Integral = (i: I) => i.toDouble.map(v => math.round(v)).toIntegral
+}
+
+
+/**
+ * Exp transformer: returns Euler's number `e` raised to the power of feature value
+ *
+ * @param uid      uid for instance
+ * @param tti      type tag for input
+ * @tparam I       input feature type
+ */
+class ExpTransformer[I <: OPNumeric[_]]
+(
+  uid: String = UID[ExpTransformer[_]]
+)(
+  implicit override val tti: TypeTag[I]
+) extends UnaryTransformer[I, Real](operationName = "exp", uid = uid){
+  override def transformFn: I => Real = (i: I) => i.toDouble.map(math.exp).filter(Number.isValid).toReal
+}
+
+
+/**
+ * Square root transformer
+ *
+ * @param uid      uid for instance
+ * @param tti      type tag for input
+ * @tparam I       input feature type
+ */
+class SqrtTransformer[I <: OPNumeric[_]]
+(
+  uid: String = UID[ExpTransformer[_]]
+)(
+  implicit override val tti: TypeTag[I]
+) extends UnaryTransformer[I, Real](operationName = "sqrt", uid = uid){
+  override def transformFn: I => Real = (i: I) => i.toDouble.map(math.sqrt).filter(Number.isValid).toReal
+}
+
+/**
+ * Log base N transformer
+ *
+ * @param base     base for log value
+ * @param uid      uid for instance
+ * @param tti      type tag for input
+ * @param n        value converter
+ * @tparam I       input feature type
+ * @tparam N       value type
+ */
+class LogTransformer[I <: OPNumeric[_], N]
+(
+  val base: N,
+  uid: String = UID[LogTransformer[_, _]]
+)(
+  implicit override val tti: TypeTag[I],
+  val n: Numeric[N]
+) extends UnaryTransformer[I, Real](operationName = "log", uid = uid){
+  override def transformFn: I => Real = (i: I) => {
+    def logN(v: Double): Double = math.log10(v) / math.log10(n.toDouble(base))
+    i.toDouble.map(logN).filter(Number.isValid).toReal
+  }
+}
+
+
+
+/**
+ * Power transformer
+ *
+ * @param base     base for log value
+ * @param uid      uid for instance
+ * @param tti      type tag for input
+ * @param n        value converter
+ * @tparam I       input feature type
+ * @tparam N       value type
+ */
+class PowerTransformer[I <: OPNumeric[_], N]
+(
+  val power: N,
```

Comment:
let's add validation that `power` is within appropriate limits

Reply:
I think `power` can be just `Double`

Reply:
and what are those limits?

Reply:
for example - do we want to allow NaN? or do we want to allow negative values?

Reply:
negatives are valid...